Formula/g/ghc.rb

```diff
@@ -46,6 +46,11 @@ class Ghc < Formula
   uses_from_macos "m4" => :build
   uses_from_macos "ncurses"
 
+  # Build uses sed -r option, which is not available in Catalina shipped sed.
+  on_catalina do
```

Comment:
I'm guessing this needs to be Catalina or earlier

Reply:
I am not completely sure but `depends_on macos: :catalina` was added in https://github.com/Homebrew/homebrew-core/commit/a46fa896dd9d474b165a9b99aeb61eecaf3df0cf and then removed recently in https://github.com/Homebrew/homebrew-core/commit/a6f51a1bdabc74c45f294ef2061e42ada52b226f. My previous [commit](https://github.com/Homebrew/homebrew-core/commit/92ddef2994d6ef3d9136f6ee3ede5594622f2487) assumed the removed line was valid. I guess I should have re-added this line too?

Reply:
I assumed the line needed to re-included, so I amended the commit and re-pushed.

Reply:
The thing is, we don't support Catalina anymore. So this code will never be tested on our side. This is why I removed it.

Reply:
Sort of both. We could keep the `sed` accommodation maintained for a few months. Maybe drop it completely for ghc 9.8 (when we release it) and keep it in ghc 9.6? It is not a big deal.

I guess you are compiling `ghc` from source then, as we don't ship a bottle anymore? I recommend that you try to upgrade as soon as possible :)

Reply:
I am compiling from source. I cannot update as my machine is too old!

Reply:
@iMichka @SMillerDev do you need any more input from me?

I will note that are still many (100+) Catalina checks in the repository.